### PR TITLE
Fixed Charge Calculation in RawData and TrapSimu

### PR DESCRIPTION
**Pull request by f3sch**
While reading the source code I found more miscalculated bitshifts.
Therefore I slightly altered the Interface for RawData to make it more
robust (easier).

This is directly related and explained in #8763.
Is somewhat related to #8778.

### CODE REVIEW

File: DataFormats/Detectors/TRD/src/RawData.cxx [hunk 1]
```diff
@@ -66,19 +66,33 @@ uint16_t buildTRDFeeID(int supermodule, int side, int endpoint)
...
+   *                           -> Q2 is 6 bits                    -> shift 5 the right for upper 2 bits
+   */
 }
 
 uint32_t getlinkerrorflag(const HalfCRUHeader& cruhead, const uint32_t link)
 {
   // link is the link you are requesting information on, 0-14
   uint32_t errorflag = 0;
-  //dealing with word0-2
+  // dealing with word0-2
```
Comment (f3sch) [inline DataFormats/Detectors/TRD/src/RawData.cxx:96]: this somehow formated...

File: Detectors/TRD/simulation/src/TrapSimulator.cxx [hunk 2]
```diff
@@ -746,28 +746,24 @@ int TrapSimulator::packData(std::vector<uint32_t>& rawdata, uint32_t offset) con
...
         case 1:
-          mcmhead.pid1 = ((mTrackletArray64[1].getQ2()) << 2) + ((mTrackletArray64[1].getQ1()) >> 5);
+          mcmhead.pid1 = getTrackletMCMHeaderQ(mTrackletArray64[1]);
           break; // all of Q2 and upper 2 bits of Q1.
         case 2:
-          mcmhead.pid2 = ((mTrackletArray64[2].getQ2()) << 2) + ((mTrackletArray64[2].getQ1()) >> 5);
+          mcmhead.pid2 = getTrackletMCMHeaderQ(mTrackletArray64[2]);
           break; // all of Q2 and upper 2 bits of Q1.
       }
-      tracklets[i].checkbit = 1;
```
Comment (f3sch) [inline Detectors/TRD/simulation/src/TrapSimulator.cxx:765]: this is redundant as in line 770 we set it to 0

File: Detectors/TRD/simulation/src/TrapSimulator.cxx [hunk 3]
```diff
@@ -746,28 +746,24 @@ int TrapSimulator::packData(std::vector<uint32_t>& rawdata, uint32_t offset) con
...
       LOG(debug) << mTrackletArray64[i];
-      tracklets[i].pid = mTrackletArray64[i].getQ0() & (mTrackletArray64[i].getQ1() << 8);
```
Comment (f3sch) [inline Detectors/TRD/simulation/src/TrapSimulator.cxx:767]: incorrect calculation of pid

**Comment by martenole** [comment]
Hi @bazinski are these changes fine with you? If merged, there will be a (small) conflict with https://github.com/AliceO2Group/AliceO2/pull/8778

**Comment by bazinski** [comment]
@martenole yes barring the fact that I wanted to deprecate all the "build" stuff in rawdata.[cxx|h] but I can still do that I suppose.
Once I find the missing tracklets 8878 is done and includes new raw readers changes to rawdata, but will just rebase to this.

File: DataFormats/Detectors/TRD/src/RawData.cxx (hunk 1 above)
Comment (martenole) [reply DataFormats/Detectors/TRD/src/RawData.cxx:96]: That happens automatically due to the clang formatting rules. Maybe the line originally did not pass the formatting checks. So I wouldn't worry about this

**Review by martenole — DISMISSED** [review]
Looks good to me, so let's see what the CI says

**Comment by f3sch** [comment]
This PR is marked as draft for now. But should be merged, if #8862 is peer-reviewed as correct.

**Pull request closed without merging**